docs/sequence/wildcard-pattern-matching.hpp

```diff
@@ -0,0 +1,30 @@
+---
+title: ワイルドカードパターンマッチング
+documentation_of: //src/sequence/wildcard-pattern-matching.hpp
+---
+
+## wildcard_pattern_matching
+```
+template< modint, class T, class Iter >
+std::vector< i32 >
+wildcard_pattern_matching(Iter f1, Iter l1, Iter f2, Iter l2, T wildcard)
+```
+
+イテレータ $[f1, l1)$, $[f2, l2)$ のワイルドカードありパターンマッチングの結果を返す。
+
+result の配列の $i$ 番目には $[f1 + i, f1 + i + (l2 - f2))$ と $[f2, l2)$ がマッチしている場合には 1 が、マッチしていない場合は 0 が格納されている。
+
+`wildcard` にはワイルドカードと扱うべき値を渡す。
+
+`T` と `Iter` についてはおそらく推論が効くが、`modint` については効かないと思われるため明示するとよい。
+
+### 計算量
+- $n = l1 - f1, m = l2 - f2$ として $O((n + m) \log (n + m))$
```

Comment:
$m \le n$ を仮定できるので

```suggestion
- $n = l1 - f1, m = l2 - f2$ として $O(n \log n)$
```

としてもよいきがする

Reply:
これ、ei1333 って人によると $O(n \log m)$ らしいんだけどあんまり考えてない

Reply:
そのei1333って人、誰なんだ…

Reply:
nlogmにするためには実装を工夫する必要があるらしい

Reply:
じゃあいいか